src/lib.rs

```diff
@@ -327,11 +311,11 @@ impl ElasticClient for hyper::Client {
 
         match req.method {
             HttpMethod::Get => self.get(&url).headers(params.headers.to_owned()).send(),
-            HttpMethod::Post => req_with_body!(self, url, req, params, post),
+            HttpMethod::Post => self.post(&url).headers(params.headers.to_owned()).body(&**req.body.unwrap().into_owned()).send(),
```

Comment:
Hmm, it'd be nice if we could avoid copying the `body` multiple times. Does `reqwest` then go and copy this owned `Vec` again into some request buffer? I'd originally used a `Cursor` over the borrowed body to avoid copying those bytes onto the heap.

Maybe it's not actually a thing though and I'm prematurely optimising. What do you think @stephanbuys?